src/canvas/PatternPath.tsx

```diff
@@ -94,4 +99,49 @@ export class PatternPath implements IPatternPath {
         const midPoint = this._computeMiddlePoint(prevPoint, currPoint);
         this._path2D.quadraticCurveTo(prevPoint.getX(), prevPoint.getY(), midPoint.getX(), midPoint.getY());
     }
+
+    private _squaredDistance = (point1: Point, point2: Point): number => {
+        const dx = point2.getX() - point1.getX();
+        const dy = point2.getY() - point1.getY();
+        return dx*dx + dy*dy;
+    }
+
+    // Returns a selection of points from the _points array that sum up the points
+    private _selectPoints = (): Point[] => {
```

Comment:
Maybe we can all this _selectPointsForSmoothing so it is more specific.

Reply:
ok